File: challenges/ttl-store/typescript/starter/exercise.ts
import type { Clock, KeyValueStore } from "../../../../tooling/typescript/contracts.js";

// Implement the challenge contract here.
export class TTLStore implements KeyValueStore {
  constructor(clock: Clock) {
    throw new Error("Not implemented: constructor");
  }

  put(key: string, value: string, ttlMs: number): void {
    throw new Error("Not implemented: put");
  }

  get(key: string): string | null {
    throw new Error("Not implemented: get");
  }

  delete(key: string): boolean {
    throw new Error("Not implemented: delete");
  }
}

File: challenges/ttl-store/python/starter/exercise.py
"""Implement the contract in the challenge README. Reference code is separate."""
from tooling.python.contracts import Clock


class TTLStore:

    def __init__(self, clock: Clock):
        raise NotImplementedError('Not implemented: __init__')

    def put(self, key: str, value: str, ttl_ms: int) -> None:
        raise NotImplementedError('Not implemented: put')

    def get(self, key: str) -> str | None:
        raise NotImplementedError('Not implemented: get')

    def delete(self, key: str) -> bool:
        raise NotImplementedError('Not implemented: delete')
